examples/plot_held_out_enet.py

```diff
@@ -0,0 +1,141 @@
+"""
+============================
+Lasso with held-out test set
+============================
+
+This example shows how to perform hyperparameter optimization
+for an elastic-net using a held-out validation set.
+
+"""
+
+# Authors: Quentin Bertrand <quentin.bertrand@inria.fr>
+#          Quentin Klopfenstein <quentin.klopfenstein@u-bourgogne.fr>
+#
+# License: BSD (3-clause)
+
+import time
+from sparse_ho.datasets.real import get_data
+import numpy as np
+from sklearn.linear_model import ElasticNet as ElasticNet_sk
+from numpy.linalg import norm
+import matplotlib.pyplot as plt
+from mpl_toolkits.mplot3d import Axes3D
+from sklearn.datasets import make_regression
+from sklearn.model_selection import train_test_split
+from sparse_ho.implicit_forward import ImplicitForward
+from sparse_ho.criterion import CV
+from sparse_ho.models import ElasticNet
+from sparse_ho.ho import grad_search
+from sparse_ho.utils import Monitor
+
+Axes3D  # hack for weird matplotlib bug
+
+# dataset = "rcv1"
+dataset = 'simu'
+# use_small_part = False
+use_small_part = True
+
+#############################
+print("Started to load data")
+
+if dataset == 'rcv1':
+    X_train, X_val, X_test, y_train, y_val, y_test = get_data(dataset)
+else:
+    X, y, beta = make_regression(n_samples=100, n_features=1000, noise=3.0, coef=True, n_informative=10)
+    X = (X - X.min(axis=0)) / (X.max(axis=0) - X.min(axis=0))
+    beta = beta / norm(beta)
+    y = X @ beta + np.random.randn(X.shape[0])
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33)
+    X_train, X_val, y_train, y_val = train_test_split(
+        X_train, y_train, test_size=0.5)
+# X_train, X_val, X_test, y_train, y_val, y_test = get_data(dataset)
+# if use_small_part:
+#     idx = np.abs((X_train.T @ y_train)).argsort()[-1000:]
+#     X_train = X_train[:, idx]
+#     X_val = X_val[:, idx]
+#     X_test = X_test[:, idx]
+
+print("Finished loading data")
+
+alpha_max = np.max(np.abs(X_train.T @ y_train))
```

Comment:
for nicer rendering of examples with sections use a line like this:

############################################################